seismicpro/src/refractor_velocity/refractor_velocity.py

```diff
@@ -0,0 +1,572 @@
+"""Implements RefractorVelocity class for estimating the velocity model of an upper part of the section."""
+
+from functools import partial
+
+import numpy as np
+from sklearn.linear_model import SGDRegressor
+from scipy import optimize
+
+from ..decorators import plotter
+from ..utils import set_ticks, set_text_formatting
+from ..utils.interpolation import interp1d
+
+
+# pylint: disable=too-many-instance-attributes, protected-access
+class RefractorVelocity:
+    """The class stores and fits parameters of a velocity model of an upper part of the section.
+
+    The class could be created from `from_first_breaks`, `from_params`, and `from_constant_velocity` classmethods.
+        * `from_first_breaks` - creates a RefractorVelocity instance, fits the parameters of the velocity model by
+                                the offsets, first break times, estimate parameters of the velocity model and stores
+                                the fitted parameters.
+        * `from_params` - creates a RefractorVelocity instance and stores the given parameters of the velocity model.
+        * `from_constant` - creates a RefractorVelocity instance based on the 1-layer velocity model with zero
+                            intercept time and given velocity. Stored parameters of the velocity model in `params`.
+
+    Some of class attributes and classmethod's arguments have a valid key notation.
```

Comment:
```suggestion
    Parameters of the constructed velocity model can be obtained by accessing the following attributes:
```
You often use "valid key notation" without even defining what it is. I'd remove it at all since it confuses more than helps.

Reply:
I often use "valid key notation" and explain what that means in a separate block in the class's docstring.
It may bea good idea to add "valid key notation" to the head of this block to help the users find this block.

Reply:
The explanation is unclear, just apply suggestion below this block.